### PR TITLE
Console: Update to angular 16

### PR DESCRIPTION
**Pull request by ThibaudAV**
## Issue

n/a

## Description

- Update Angular core
- Update Angular material
- Use InjectionToken for Constants inject
- rework guard to match new way to do

⚠️ ⚠️  After this PR we need to use Constants token
- Replace `@Inject('Constants') readonly constants: Constants,` by `@Inject(Constants) readonly constants: Constants,`

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-pqxfxjajop.chromatic.com)

### CODE REVIEW

File: .circleci/ci/src/workflows/workflow-pull-requests.ts [hunk 1]
```diff
@@ -223,7 +223,7 @@ export class PullRequestsWorkflow {
           name: 'Lint & test APIM Console',
           context: config.jobContext,
           'apim-ui-project': config.dockerImages.console.project,
-          resource_class: 'large',
+          resource_class: 'xlarge',
```
Comment (ThibaudAV) [inline .circleci/ci/src/workflows/workflow-pull-requests.ts:226]: From my analysis we had a Job with between 15 and 25 minutes before. With the changes of this commit and the xlarge we are more on an execution time of ~8 9 minutes. 
This divides the time by 2 and should not change the use of credit (as an xlarge is 2 times more expensive than a large) 💭

File: gravitee-apim-console-webui/jest.config.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 process.env.TZ = 'UTC';
 
 module.exports = {
+  workerIdleMemoryLimit: '896MB',
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/jest.config.js]: 📝  for myself : Try to remove this when we use node 20.x 
Deep subject here : https://github.com/jestjs/jest/issues/ 11956

File: .circleci/ci/src/workflows/workflow-pull-requests.ts (hunk 1 above)
Comment (ThibaudAV) [reply .circleci/ci/src/workflows/workflow-pull-requests.ts:226]: I think there will be some flaky left, but that's already better.

**Review by Okhelifi — APPROVED** [review]

File: gravitee-apim-console-webui/package.json [hunk 3]
```diff
@@ -22,8 +22,8 @@
     "@fontsource/libre-franklin": "4.4.5",
     "@gravitee/ui-analytics": "9.6.0",
     "@gravitee/ui-components": "4.1.1",
-    "@gravitee/ui-particles-angular": "9.6.0",
-    "@gravitee/ui-policy-studio-angular": "9.6.0",
+    "@gravitee/ui-particles-angular": "^9.6.0-angular-16-615de4d",
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/package.json:25]: We're merging with a temporary branch while we wait for cockpit validation, which will arrive shortly.

**Pull request merged**